### DISCUSSION TITLE
问一些关于本项目的疑问

### DISCUSSION THREAD

**Discussion by shao4598** [msojocs/wechat-web-devtools-linux#119] [Q&A]
- 运行程序时，中文字体显示成方块乱码。作者可否提供一份这个程序的中文字体清单，我们可以按照清单自行安装字体。
- README中提到的“wine版本”是什么意思，我没有在Release中看到相关信息，可否在README给出具体的描述。
- 这个项目的原理，是不是把需要wine执行的.dll，放到了Windows Docker中执行？希望作者可以把这部分内容做下具体描述，供我们参考学习。
- README中给出了源码构建教程。请问有什么场景需要这么做？或者说在什么场景下，源码构建可以获得更好的开发体验？
- 希望作者可以把.rpm发行方式加到Release中，如果作者精力有限，我可以协助完成这部分工作。
- 最后给作者点赞👍，我觉得这个项目是全网同类型项目中，安装最方便、更新最及时、使用体验最好的项目了，希望大家能够携手把这个项目维持下去。

**Comment by msojocs** [comment]
1. 方块不是乱码，是字体没找到，目前已知是装微软雅黑能解决一些，但似乎不全。（#92）
2. wine 是早期编译器依赖于wine环境，后面有Linux原生编译器后就不再维护了。
3. 没用windows docker，都是原生linux程序。
4. 源码构建没什么优点，是给想要PR的人看的。
5. rpm 有空在看能不能加。

**Reply by shao4598** [reply]
![截图 2024-09-24 10-06-17](https://github.com/user-attachments/assets/544a5e99-12f4-4c2e-a54c-115c10802b83)
我本机的微软雅黑之前是已安装的，似乎并没有生效（Fedora 40， 解压tar.gz后使用二进制程序运行，如有需要我去Issues里详细描述）。

**Reply by msojocs** [reply]
字体应该有这些：

"Helvetica Neue", Helvetica, "Hiragino Sans GB", "Microsoft YaHei", 微软雅黑, Arial, sans-serif;

**Reply by shao4598** [reply]
好的，我排查出来的原因是nw.js的“font-family: sans-serif;“没有读取到Linux操作系统的无衬线字体。
接下来我会自己尝试运行一个nw.js的Demo进一步排查这个问题。可能我最终会修复Linux操作系统的字体配置，或者修改开发者工具源码重新构建。
等有了结果我再在下面做回复。

**Reply by shao4598** [reply]
我有解决方案了。
原因是 Fedora 的 sans-serif 默认不是中文字体。
解决方案如下：
1. 新建文件 ```~/.config/fontconfig/fonts.conf```。
2. 按如下配置默认字体，```<string/>``` 里的字体按个人喜好选择，只要保证是中文字体即可。
```xml
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
  <match>
    <edit name="family" mode="prepend">
      <string>PingFang SC</string>
    </edit>
  </match>
</fontconfig>
```
3. 保存并退出fonts.conf。
4. 刷新字体配置缓存。
```shell
$ fc-cache -f -v
```
